alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -0,0 +1,253 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.workflow.processor;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.enumeration.FormatType;
+import com.synopsys.integration.alert.common.enumeration.ItemOperation;
+import com.synopsys.integration.alert.common.message.model.AggregateMessageContent;
+import com.synopsys.integration.alert.common.message.model.CategoryItem;
+import com.synopsys.integration.alert.common.message.model.CategoryKey;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+
+@Component
+public class SummaryMessageContentProcessor extends MessageContentProcessor {
+    private final DigestMessageContentProcessor digestMessageContentProcessor;
+
+    @Autowired
+    public SummaryMessageContentProcessor(final DigestMessageContentProcessor digestMessageContentProcessor) {
+        super(FormatType.SUMMARY);
+        this.digestMessageContentProcessor = digestMessageContentProcessor;
+    }
+
+    @Override
+    public List<AggregateMessageContent> process(final List<AggregateMessageContent> messages) {
+        final List<AggregateMessageContent> digestMessages = digestMessageContentProcessor.process(messages);
+        return digestMessages
+                   .stream()
+                   .map(this::summarize)
+                   .collect(Collectors.toList());
+    }
+
+    private AggregateMessageContent summarize(final AggregateMessageContent message) {
+        final SortedSet<CategoryItem> originalCategoryItems = message.getCategoryItems();
+        if (null == originalCategoryItems) {
+            return message;
+        }
+
+        final Map<ItemOperation, SortedSet<CategoryItem>> itemsByOperation = sortByOperation(originalCategoryItems);
+
+        final SortedSet<CategoryItem> summarizedCategoryItems = new ConcurrentSkipListSet<>();
```

Comment:
Why did you choose ConcurrentSkipListSet over TreeSet? We use TreeSet everywhere else as the type of SortedSet